Add method CreateStreamFromMemory to get stream in-memory

stanleyycho opened the pull request:
Hello, 

I have an use case that I need to read attachment in-memory. There seem no other ways to read the attached images without saving it to file first. So I implemented this function `CreateStreamFromMemory` to copy the stream directly. The usage is the following.
```
std::shared_ptr<libCZI::IStream> cziStream_ = libCZI::CreateStreamFromFile(L"CZI.czi");
std::shared_ptr<libCZI::ICZIReader> reader_ = libCZI::CreateCZIReader();
reader_->Open(cziStream_);
std::shared_ptr<libCZI::ICZIReader> labelReader = libCZI::CreateCZIReader();
std::shared_ptr<libCZI::IStream> strm;
reader_->EnumerateAttachments(
	[&](int index, const libCZI::AttachmentInfo& info)->bool
{
	auto attchmnt = reader_->ReadAttachment(index);
	if (!attchmnt)
	{
		return false;
	}
	if (info.name == "Label")
	{
		size_t size;
		std::shared_ptr<const void> spData = attchmnt->GetRawData(&size);
		// Create reader stream in memory
		strm = libCZI::CreateStreamFromMemory(spData.get());
		labelReader->Open(strm);
		auto tileAccessor = labelReader->CreateSingleChannelTileAccessor();
	}
}
```

ptahmose commented:
Thanks, this is a useful addition!
I'd suggest that
- the memory-stream-object should take ownership of the memory it operates on (e.g. by passing in the shared-pointer instead of the raw pointer)
- it should not be able to read past the end of the buffer

stanleyycho commented:
Hi @ptahmose, 

Thank you for the feedback. Sorry for the late response.
I updated the raw pointer and added one more constructor to take attachment instead.
```
strm = libCZI::CreateStreamFromMemory(attchmnt);
```

zeissmicroscopy commented:
Added your changeset with some minor changes - thanks again!
